clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -385,7 +392,9 @@ public void verifyHeartbeatSentWhenFetchedDataReady() throws Exception {
         consumer.subscribe(singleton(topic), getConsumerRebalanceListener(consumer));
         Node coordinator = prepareRebalance(client, node, assignor, singletonList(tp0), null);
 
-        consumer.poll(0);
+        consumer.internalUpdateAssignmentMetadataIfNeeded(Duration.ZERO);
+        consumer.internalUpdateAssignmentMetadataIfNeeded(Duration.ZERO);
+        consumer.poll(Duration.ZERO);
```

Comment:
We need to issue the poll request because the mock is expecting it, but we also need the double-metadata call to initialize the metadata beforehand, otherwise the internal code won't issue the fetch request.